DungeonWorld/DAO/DataAccesObject.cs

```diff
@@ -1,62 +1,26 @@
 ﻿using DungeonWorld.Model;
 using System;
 using System.Collections.Generic;
-using System.Data.Entity;
 using System.Linq;
-using System.Threading.Tasks;
 
 namespace DungeonWorld.DAO
 {
     public class DataAccesObject
     {
         public static DWModel context = new DWModel();
```

Comment:
Najgorsze co mogłeś zrobić.

DbContext nie jest Thread Safe to znaczy, jakbyś programował aplikację wielowątkową (np. apkę webową) to wiele tasków/wątków próbowałaby się dostać do twojego contextu (a w zasadzie jego instancji).

Context to taka abstrakcja nałożona na połączenie z bazą danych która nie współgra z wieloma Taskami. Jedno połączenie == jeden wątek jeden task.

Jeśli tak bardzo chciałbyś mieć context statyczny przy każdym query mógłbyś robić coś takiego:

```csharp
lock (context)
{
    context.Races.Where(r => r.IDRace == id).FirstOrDefault();
}
```

`lock` powoduje, że do contextu aktualnie dostęp ma tylko jeden wątek. Lock się zwalnia, kiedy wątek kończy pracę. Ogólnie static w tej klasie to bardzo niefortunny modyfikator dostępu. Najlepiej w ogóle nie używaj staticów. 

Przemyśl o dodaniu do tego projektu jakiegoś IoC/frameworka Di. Np. `Autofac`. Wtedy context mógłbyś w łatwy sposób wstrzykiwać do klas.